images/virtualization-artifact/pkg/controller/vmsnapshot/internal/life_cycle.go

```diff
@@ -462,7 +462,7 @@ func (h LifeCycleHandler) ensureBlockDeviceConsistency(ctx context.Context, vm *
 			return fmt.Errorf("%w: waiting for the Ready condition of the virtual disk %q to be True", ErrVirtualDiskResizing, vd.Name)
 		}
 
-		resizingReady, _ := conditions.GetCondition(vdcondition.ResizedType, vd.Status.Conditions)
+		resizingReady, _ := conditions.GetCondition(vdcondition.ResizingType, vd.Status.Conditions)
```

Comment:
It should see not a reason, but a status of condition now.
Add please check for condition observed generation as well.